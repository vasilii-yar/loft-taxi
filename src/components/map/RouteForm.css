.route-form-layout {
    padding: 44px 60px;
    margin-top: 80px;
    margin-bottom: 20px;
    top: 0;
    left: 20px;
    position: absolute;
    width: 300px;
    height: 170px;
}