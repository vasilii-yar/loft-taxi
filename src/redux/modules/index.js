import authReducer from "./auth/authReducer";
import profileReducer from "./profile/profileReducer";
import {combineReducers} from "redux";


export default combineReducers({
    auth: authReducer,
    profile: profileReducer
});