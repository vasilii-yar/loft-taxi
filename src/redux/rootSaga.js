import {all} from "@redux-saga/core/effects";
import {authSaga} from "./modules/auth/authSaga";
import {profileSaga} from "./modules/profile/profileSaga";
import {addressSaga} from "./modules/address/addressSaga";
import {routeSaga} from "./modules/route/routeSaga";

export default function* rootSaga() {
    yield all([
        authSaga(),
        profileSaga(),
        addressSaga(),
        routeSaga()
    ])
}