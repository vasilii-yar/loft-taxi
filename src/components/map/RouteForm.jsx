import React from 'react';
import Paper from "@material-ui/core/Paper";
import {Box} from "@material-ui/core";
import Grid from "@material-ui/core/Grid";
import Button from "@material-ui/core/Button";
import Autocomplete from "@material-ui/lab/Autocomplete"
import TextField from "@material-ui/core/TextField";
import "./RouteForm.css";
import {connect} from "react-redux";
import {tryFetchAddress} from "../../redux/modules/address/addressActions";
import {tryRouting} from "../../redux/modules/route/routeActions";

class RouteForm extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            fromValue: "",
            toValue: "",
            inputValid: true,
            errorText: "",
            isRoutePrinted: false
        }
    }

    handleSubmit = (event) => {
        event.preventDefault();
        if (this.state.isRoutePrinted) {
            this.setState({
                fromValue: "",
                toValue: "",
                isRoutePrinted: false
            })
        } else if (this.validate()) {
            this.props.tryRouting(this.state.fromValue, this.state.toValue);
            this.setState({
                isRoutePrinted: true
            });
        }
    }

    handleChange = (event, newValue) => {
        const {id} = event.target;
        const name = (id.indexOf("from-combo") === -1) ? "toValue" : "fromValue";
        this.setState({[name]: newValue});
    }

    validate = () => {
        const from = this.state.fromValue;
        const  to = this.state.toValue;

        if (from === "" || to === "") {
            this.setState({
                inputValid: false,
                errorText: "Ошибка! Заполните все поля!"
            })
            return false;
        }

        if (from === to) {
            this.setState({
                inputValid: false,
                errorText: "Ошибка! Значения должны отличаться!"
            })
            return false;
        }

        if (!this.state.inputValid) {
            this.setState({
                inputValid: true,
                errorText: ""
            })
        }

        return true;
    }

    componentDidMount() {
        this.props.tryFetchAddress();
    }

    render() {
        return (
            <Paper className="route-form-layout">
                <Box>
                    <form onSubmit={this.handleSubmit} className="route-form-content">
                        <Grid container direction="column" spacing={4}>
                            <Grid item xs={12}>
                                <Autocomplete
                                    id="from-combo"
                                    options={this.props.addressList}
                                    getOptionLabel={(option) => option}
                                    value={this.state.fromValue}
                                    onChange={this.handleChange}
                                    renderInput={(params) => <TextField {...params} label="Откуда"
                                                                        variant="standard"
                                                                        error={!this.state.inputValid}
                                                                        helperText={this.state.errorText}/>}
                                />
                            </Grid>
                            <Grid item xs={12}>
                                <Autocomplete
                                    id="to-combo"
                                    options={this.props.addressList}
                                    getOptionLabel={(option) => option}
                                    value={this.state.toValue}
                                    onChange={this.handleChange}
                                    renderInput={(params) => <TextField {...params} label="Куда"
                                                                        variant="standard"
                                                                        error={!this.state.inputValid}
                                                                        helperText={this.state.errorText}/>}
                                />
                            </Grid>
                            <Grid item xs={12}>
                                <div className="get-taxi-button">
                                    <Button variant="contained" color="primary" type="submit" fullWidth>
                                        {this.state.isRoutePrinted ? "Сбросить маршрут" : "Вызвать такси"}
                                    </Button>
                                </div>
                            </Grid>
                        </Grid>
                    </form>
                </Box>
            </Paper>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        addressList: state.address.addressList
    }
}

export default connect(
    mapStateToProps,
    {tryFetchAddress, tryRouting}
)(RouteForm);