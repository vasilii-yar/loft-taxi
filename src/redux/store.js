import {applyMiddleware, createStore} from "redux";
import mainReducer from "./modules"
import {authMiddleware} from "./middleware/authMiddleware";
import {profileMiddleware} from "./middleware/profileMiddleware";

export const store = createStore(mainReducer, applyMiddleware(authMiddleware, profileMiddleware));