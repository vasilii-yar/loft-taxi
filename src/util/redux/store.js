import {applyMiddleware, createStore} from "redux";
import mainReducer from "../../reducers"
import {authMiddleware} from "../../middleware/authMiddleware";

export const store = createStore(mainReducer, applyMiddleware(authMiddleware));