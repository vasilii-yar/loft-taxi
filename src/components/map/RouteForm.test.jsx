import React from 'react';
import {render} from "@testing-library/react";
import {BrowserRouter} from "react-router-dom";
import {Provider} from "react-redux";
import {store} from "../../redux/store";
import renderer from "react-test-renderer";
import RouteForm from "./RouteForm";

describe("RouteForm", () => {
    it("renders correctly", () => {
        const {container} = render(
            <BrowserRouter>
                <Provider store={store}>
                    <RouteForm/>
                </Provider>
            </BrowserRouter>
        );
        expect(container.innerHTML).toMatch("Вызвать такси");
    })

    it("renders correctly check by snapshot", () => {
        const tree = renderer.create(
            <BrowserRouter>
                <Provider store={store}>
                    <RouteForm/>
                </Provider>
            </BrowserRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })
})