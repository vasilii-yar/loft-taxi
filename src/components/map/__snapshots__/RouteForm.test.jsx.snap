// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouteForm renders correctly check by snapshot 1`] = `
<div
  className="MuiPaper-root route-form-layout MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiBox-root MuiBox-root-1"
  >
    <form
      className="route-form-content"
      onSubmit={[Function]}
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-direction-xs-column"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            aria-expanded={false}
            aria-owns={null}
            className="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="combobox"
          >
            <div
              className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                data-shrink={false}
                htmlFor="from-combo"
                id="from-combo-label"
              >
                Откуда
              </label>
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
                onClick={[Function]}
              >
                <input
                  aria-activedescendant={null}
                  aria-autocomplete="list"
                  aria-controls={null}
                  aria-invalid={false}
                  autoCapitalize="none"
                  autoComplete="off"
                  autoFocus={false}
                  className="MuiInputBase-input MuiInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd"
                  disabled={false}
                  id="from-combo"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                  required={false}
                  spellCheck="false"
                  type="text"
                  value=""
                />
                <div
                  className="MuiAutocomplete-endAdornment"
                >
                  <button
                    aria-label="Clear"
                    className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator MuiAutocomplete-clearIndicatorDirty"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={-1}
                    title="Clear"
                    type="button"
                  >
                    <span
                      className="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                        />
                      </svg>
                    </span>
                  </button>
                  <button
                    aria-label="Open"
                    className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={-1}
                    title="Open"
                    type="button"
                  >
                    <span
                      className="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M7 10l5 5 5-5z"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
              </div>
              
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            aria-expanded={false}
            aria-owns={null}
            className="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="combobox"
          >
            <div
              className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                data-shrink={false}
                htmlFor="to-combo"
                id="to-combo-label"
              >
                Куда
              </label>
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
                onClick={[Function]}
              >
                <input
                  aria-activedescendant={null}
                  aria-autocomplete="list"
                  aria-controls={null}
                  aria-invalid={false}
                  autoCapitalize="none"
                  autoComplete="off"
                  autoFocus={false}
                  className="MuiInputBase-input MuiInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd"
                  disabled={false}
                  id="to-combo"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                  required={false}
                  spellCheck="false"
                  type="text"
                  value=""
                />
                <div
                  className="MuiAutocomplete-endAdornment"
                >
                  <button
                    aria-label="Clear"
                    className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator MuiAutocomplete-clearIndicatorDirty"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={-1}
                    title="Clear"
                    type="button"
                  >
                    <span
                      className="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                        />
                      </svg>
                    </span>
                  </button>
                  <button
                    aria-label="Open"
                    className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={-1}
                    title="Open"
                    type="button"
                  >
                    <span
                      className="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M7 10l5 5 5-5z"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
              </div>
              
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            className="get-taxi-button"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="submit"
            >
              <span
                className="MuiButton-label"
              >
                Вызвать такси
              </span>
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
